pkg/nsx/services/health/healthchecker.go

```diff
@@ -0,0 +1,264 @@
+}
+
+// ClusterHealthChecker is the main structure for performing health checks
+type ClusterHealthChecker struct {
+	nsxClient         *nsx.Client
+	k8sClient         client.Client
+	handlers          *HealthCheckHandlers
+	healthCheckCtx    context.Context
+	healthCheckCancel context.CancelFunc
+}
+
+// NewClusterHealthChecker creates a new ClusterHealthChecker instance
+func NewClusterHealthChecker(nsxClient *nsx.Client, k8sClient client.Client) *ClusterHealthChecker {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	checker := &ClusterHealthChecker{
+		nsxClient:         nsxClient,
+		k8sClient:         k8sClient,
+		handlers:          NewHealthCheckHandlers(),
+		healthCheckCtx:    ctx,
+		healthCheckCancel: cancel,
+	}
+
+	// Register default health check handlers
+	checker.registerDefaultHandlers()
+
+	return checker
+}
+
+// registerDefaultHandlers registers the default health check handlers
+func (c *ClusterHealthChecker) registerDefaultHandlers() {
+	// NSX health check handler
+	c.handlers.AddHandler("NSX", c.checkNSXHealth)
+
+	// Kubernetes API server health check handler
+	c.handlers.AddHandler("Kubernetes", c.checkKubernetesHealth)
+
+	// TODO add _nsx_restore_check
+}
+
+// checkNSXHealth checks the health of NSX
+func (c *ClusterHealthChecker) checkNSXHealth() error {
+	return c.nsxClient.NSXChecker.CheckNSXHealth(nil)
+}
+
+// checkKubernetesHealth checks the health of Kubernetes
+func (c *ClusterHealthChecker) checkKubernetesHealth() error {
+	if c.k8sClient == nil {
+		return fmt.Errorf("kubernetes client not initialized")
+	}
+
+	// Simple health check by listing namespaces
+	namespaceList := &corev1.NamespaceList{}
+	err := c.k8sClient.List(context.TODO(), namespaceList)
+	return err
+}
+
+// CheckClusterHealth performs a one-time health check and returns the overall status
+func (c *ClusterHealthChecker) CheckClusterHealth() HealthStatus {
+	handlers := c.handlers.GetHandlers()
+	hasErrors := false
+
+	for checkItem, checkHandler := range handlers {
+		if err := checkHandler(); err != nil {
+			log.V(1).Info("Health check failed", "component", checkItem, "error", err)
+			hasErrors = true
+		}
+	}
+
+	if hasErrors {
+		return HealthStatusDown
+	}
+	return HealthStatusHealthy
+}
+
+// SystemHealthReporter reports system health status to NSX
+type SystemHealthReporter struct {
+	healthChecker  *ClusterHealthChecker
+	nsxClient      *nsx.Client
+	clusterID      string
+	reportCtx      context.Context
+	reportCancel   context.CancelFunc
+	reportInterval time.Duration
+	ticker         *time.Ticker
+}
+
+// NewSystemHealthReporter creates a new SystemHealthReporter instance
+func NewSystemHealthReporter(nsxClient *nsx.Client, clusterID string, k8sClient client.Client) *SystemHealthReporter {
+	ctx, cancel := context.WithCancel(context.Background())
+	return &SystemHealthReporter{
+		healthChecker:  NewClusterHealthChecker(nsxClient, k8sClient),
+		nsxClient:      nsxClient,
+		clusterID:      clusterID,
```

Comment:
@TaoZou1 : do we need to generate uuid of clusterID?

Reply:
fixed